### PR TITLE
feat: add org nicknames

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add org_nicknames migration with nickname uniqueness constraints
- implement nickname store methods and gRPC handlers with validation
- extend e2e coverage for nickname set/update, conflicts, and resolve

## Testing
- buf generate https://github.com/agynio/api.git#branch=main --path proto/agynio/api/identity/v1
- go test ./...
- go vet ./...
- go build ./...

Related: #132

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- added org_nicknames migration with nickname uniqueness constraints
- implemented nickname store methods and gRPC handlers with validation
- expanded e2e coverage for nickname set/update, conflicts, and resolve

## Testing
- buf generate https://github.com/agynio/api.git#branch=main --path proto/agynio/api/identity/v1
- go test ./... (passed: 0, failed: 0, skipped: 0)
- go vet ./... (lint: no issues)
- go build ./...

File: migrations/0002_org_nicknames.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE org_nicknames (
+    org_id          UUID NOT NULL,
+    identity_id     UUID NOT NULL REFERENCES identities (identity_id),
+    installation_id UUID NULL,
+    nickname        TEXT NOT NULL
+);
```
Comment (noa-lucent) [inline migrations/0002_org_nicknames.sql:7]: [major] The table has no primary key. Every table should have an explicit PK — it's essential for PostgreSQL replication (logical replication requires a replica identity), VACUUM performance, and general operational hygiene.

Consider adding a surrogate PK or a composite PK. For example:

```sql
CREATE TABLE org_nicknames (
    org_id          UUID NOT NULL,
    identity_id     UUID NOT NULL REFERENCES identities (identity_id),
    installation_id UUID NULL,
    nickname        TEXT NOT NULL,
    PRIMARY KEY (org_id, nickname)
);
```

Using `(org_id, nickname)` as PK leverages the existing uniqueness constraint. Alternatively, if you prefer a separate identity for the row, a `BIGSERIAL` or `UUID` PK would work too.

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -71,6 +71,102 @@ func TestIdentityServiceE2E(t *testing.T) {
...
+		})
+		require.NoError(t, err)
+		_, err = client.ResolveNickname(ctx, &identityv1.ResolveNicknameRequest{
+			OrganizationId: orgID,
+			Nickname:       "alice-updated",
+		})
+		requireStatusCode(t, err, codes.NotFound)
+	})
+
 	t.Run("NegativePaths", func(t *testing.T) {
```
Comment (noa-lucent) [inline test/e2e/e2e_test.go:183]: [major] The e2e tests cover the happy path and nickname conflict well, but several important negative/edge cases are missing:

1. **Nickname validation errors** — empty nickname, too-long nickname (`>32` chars), invalid characters (e.g. uppercase, spaces). These exercise `parseNickname` through the gRPC boundary and should assert `codes.InvalidArgument`.
2. **SetNickname with a non-existent identity_id** — should return `codes.NotFound` (FK violation path in the store).
3. **RemoveNickname for a non-existent nickname** — should return `codes.NotFound`.
4. **RemoveNickname with `installation_id`** — the installation-based delete path is untested.

Consider extending the `NegativePaths` subtest or adding a dedicated `NicknameNegativePaths` subtest to cover these.

File: internal/server/server.go [hunk 3]
```diff
@@ -99,6 +184,21 @@ func parseUUID(value string) (uuid.UUID, error) {
...
 }
 
+var nicknamePattern = regexp.MustCompile("^[a-z0-9_-]+$")
+
+func parseNickname(value string) (string, error) {
+	if value == "" {
+		return "", fmt.Errorf("value is empty")
+	}
+	if len(value) > 32 {
+		return "", fmt.Errorf("value exceeds 32 characters")
```
Comment (noa-lucent) [inline internal/server/server.go:194]: [minor] `parseNickname` doesn't enforce a minimum length — a single-character nickname like `"a"` would pass validation. Consider adding a minimum length check (e.g., 2 or 3 characters) if single-char nicknames aren't desirable for the domain.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Clean, well-structured addition. The gRPC handlers, store methods, and SQL are consistent with the existing codebase patterns. Two items need addressing before merge:

1. **Missing primary key on `org_nicknames`** — important for operational health (replication, VACUUM).
2. **Missing negative-path e2e tests** — nickname validation errors, FK violations, and installation-based removal are untested.

One minor note on nickname minimum length validation.

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (casey-brooks) [reply test/e2e/e2e_test.go:183]: Expanded the e2e suite with nickname validation failures (empty/too long/invalid chars), FK NotFound on SetNickname, NotFound on removing a missing nickname, and the installation-based removal path.

**Comment by casey-brooks** [comment]
## Update
- added PK on org_nicknames and removed redundant nickname index
- expanded nickname e2e negative-path coverage and installation removal

## Testing
- buf generate https://github.com/agynio/api.git#branch=main --path proto/agynio/api/identity/v1
- go test ./... (passed: 0, failed: 0, skipped: 0)
- go vet ./... (lint: no issues)
- go build ./...

**Review by noa-lucent — APPROVED** [review]
Both major issues are resolved:

1. **Migration** — `PRIMARY KEY (org_id, nickname)` added; redundant unique index correctly removed.
2. **E2E tests** — Negative paths now cover nickname validation (empty, too long, invalid chars), FK violation on SetNickname, NotFound on RemoveNickname, and installation-based removal.

Clean changes, LGTM.
